frontend/src/components/Friend.tsx

```diff
@@ -1,105 +1,104 @@
 import {
   Flex,
   Text,
-  Square,
-  Image,
-  IconButton,
   Box,
   Input,
+  IconButton,
+  Grid,
+  GridItem,
 } from "@chakra-ui/react";
-import { CheckCircleIcon, CloseIcon } from "@chakra-ui/icons";
+import { CloseIcon } from "@chakra-ui/icons";
 import { Avatar } from "@chakra-ui/react";
 import md5 from "md5";
-import { useState } from "react";
 
 export function Friend(props: {
   email: string;
   isMe?: boolean;
   isSelected?: boolean;
   onRemove?: (email: string) => void;
   splitMode?: string;
-  subscriptionCost?: number;
-  friendCount?: number;
+  splitCustomAmount: number;
```

Comment:
I this this Friend component can be "dummer". Right now it knows too much about the split mode. It should just have enough to display the amount and to change the amount, without knowing about the splitMode.

Could it be rewritten to just have the interface?
```
  email: string;
  isMe?: boolean;
  isSelected?: boolean;
  onRemove?: (email: string) => void;
  subscriptionCost?: number;
  handleSubscriptionCostChange?: (email: string, value: number) => void;
}
```

May need another bool if it can or can't be changed. (in split equally, cost can't be changed, wheras it can in split by custom amounts).